### PR TITLE
Update auto3dseg algorithm templates

### PR DESCRIPTION
**Pull request by dongyang0122**
Update auto3dseg algorithm templates for performance (accuracy) and efficiency improvement.

### CODE REVIEW

File: auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml [hunk 1]
```diff
@@ -49,10 +49,10 @@ training:
     momentum: 0.9
     weight_decay: 4.0e-05
   lr_scheduler:
-    _target_: torch.optim.lr_scheduler.StepLR
+    _target_: torch.optim.lr_scheduler.PolynomialLR
     optimizer: "$@training#optimizer"
-    step_size: "$max(@training#num_epochs // 5, 1)"
-    gamma: 0.5
+    power: 0.5,
+    total_iters: '$@training#num_epochs // @training#num_per_validation_epochs + 1'
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:55]: Thank you for the PR @dongyang0122 .

`num_per_validation_epochs` seems unreferenced to me.  Do you mean `num_epochs_per_validation`?

File: auto3dseg/algorithm_templates/dints/scripts/train.py [hunk 2]
```diff
@@ -54,8 +54,8 @@ def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
     fold = parser.get_parsed_content("fold")
     num_images_per_batch = parser.get_parsed_content("training#num_images_per_batch")
     num_epochs = parser.get_parsed_content("training#num_epochs")
-    num_epochs_per_validation = parser.get_parsed_content(
-        "training#num_epochs_per_validation"
+    num_per_validation_epochs = parser.get_parsed_content(
+        "training#num_per_validation_epochs"
```
Comment (mingxin-zheng) [inline auto3dseg/algorithm_templates/dints/scripts/train.py:58]: Should it be `num_epochs_per_validation`?

File: auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml (hunk 1 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/configs/hyper_parameters.yaml:55]: current parser cannot understand `num_epochs_per_validation` if we have already defined `num_epochs` in the .json config.

File: auto3dseg/algorithm_templates/dints/scripts/train.py (hunk 2 above)
Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/train.py:58]: current parser cannot understand `num_epochs_per_validation` if we have already defined `num_epochs` in the .json config.

**Comment by mingxin-zheng** [comment]
I would like to hear @Nic-Ma 's thought on the `ConfigParser`. Since we already use `num_epochs_per_validation` in lots of places, such as the unit tests of MONAI, tutorials of Auto3DSeg, I feel hesitated to make such change.

**Comment by Nic-Ma** [comment]
Hi @dongyang0122 @mingxin-zheng ,

I tried to reproduce the issue with below config:
```json
{
    "data": [1, 2],
    "data_per_batch": 2,
    "transform": {
        "_target_": "Compose",
        "transforms": [
            {"_target_": "LoadImaged", "keys": "image"},
            {"_target_": "RandTorchVisiond", "keys": "@##0#keys", "name": "ColorJitter", "brightness": 0.25},
        ],
    },
    "dataset": {"_target_": "Dataset", "data": "@data", "transform": "@transform"},
    "dataloader": {
        "_target_": "DataLoader",
        "dataset": "@##dataset",
        "batch_size": "@data_per_batch",
        "collate_fn": "$monai.data.list_data_collate",
    },
}
```
But seems the first 2 lines can be parsed OK:
```json
"data": [1, 2],
"data_per_batch": 2
```
Could you please help provide a simple program to reproduce the issue you asked?

Thanks.

**Comment by mingxin-zheng** [comment]
Hi @dongyang0122 , if the fix in the `dev` branch has resolved the issue of `num_epoch_*`, can we revert the name of the variable back to `num_epochs_per_validation`?

**Comment by dongyang0122** [comment]
> Hi @dongyang0122 , if the fix in the `dev` branch has resolved the issue of `num_epoch_*`, can we revert the name of the variable back to `num_epochs_per_validation`?

If we convert it back to previous name, then we have to inform users to install monai from the dev branch. I feel it adds another layer of complexity for general users. how about converting it back once the changes are reflected in the installation via `pip install monai`? and meanwhile we can merge the changes at the moment.

**Review by mingxin-zheng — APPROVED** [review]
